serverless/meetup.js

```diff
@@ -10,43 +12,106 @@ const headers = {
+const FALLBACK_IMAGE = 'https://res.cloudinary.com/codestar/image/upload/v1532409289/codestar.nl/meetup/codestar-night-logo.jpg';
 
 module.exports.getUpcomingEvents = async (event, context, callback) => {
-	// TODO origin check
-
-	// const data = await fetch(GET_UPCOMING_EVENTS_URL, {
-	// 	method: 'GET'
-	// });
-	// const resolvedData = await data.json();
-	// // const ids = resolvedData.map(meetupEvent => {
-	// // 	// TODO conditionally get event details, most importantly the image URL
-	// // 	console.log(meetupEvent.name);
-	// // 	return meetupEvent.id;
-	// // });
-	// const meetupEventPromises = resolvedData.map(meetupEvent => {
-	// 	// TODO conditionally get event details, most importantly the image URL
-	// 	console.log(meetupEvent.name);
-	// 	return fetch(`https://api.meetup.com/Code-Star-Night/events/${meetupEvent.id}?&sign=true&photo-host=public&fields=featured_photo`, {method: 'GET'});
-	// });
-	// // TODO limit to e.g. 20
-	// // TODO show placeholders
-	// const eventDetailsData = await meetupEventPromises[0]; // TODO convert to Promise.all
-	// const eventDetailsDataResolved = await eventDetailsData.json();
-	// console.log(eventDetailsDataResolved.featured_photo.photo_link);
-	// // TODO with zip?
-	// const newData = {
-	// 	id: resolvedData[0].id,
-	// 	name: resolvedData[0].name,
-	// 	photo: eventDetailsDataResolved.featured_photo.photo_link
-	// };
-	// callback(null, {
-	// 	statusCode: 200,
-	// 	headers,
-	// 	body: JSON.stringify({
-	// 		message: newData,
-	// 	}),
-	// });
+	const allowedOrigins = [allowedOrigin];
+	const debug = process.env.DEBUG;
+	if(debug === 'true') {
+		allowedOrigins.push('http://localhost:3000');
+	}
+
+	try {
+		if(!allowedOrigins.includes(event.headers.origin)) {
+			throw new Error(`Not white-listed origin: ${event.headers.origin}`);
+		}
+
+		const response = await got(GET_UPCOMING_EVENTS_URL, { json: true });
+		const mEvents = response.body.map(({ name, time, link, description, featured_photo}) => {
+			return {
+				name,
+				time,
+				link,
+				description,
+				featured_photo
+			}
+		});
+		callback(null, {
+			statusCode: 200,
+			headers,
+			body: JSON.stringify(mEvents),
+		});
+	} catch(err) {
+		callback('Failed GET_UPCOMING_EVENTS_URL ' + err);
+	}
 };
+
+module.exports.getPastEvents = async (event, context, callback) => {
+	const allowedOrigins = [allowedOrigin];
+	const debug = process.env.DEBUG;
+	if(debug === 'true') {
+		allowedOrigins.push('http://localhost:3000');
+	}
+
+	try {
+		if(!allowedOrigins.includes(event.headers.origin)) {
+			throw new Error(`Not white-listed origin: ${event.headers.origin}`);
+		}
+
+		const response = await got(GET_PAST_EVENTS_URL, { json: true });
+		const mEventsPromises = await response.body
+			.map(({ name, time, link, featured_photo}) => {
+				return {
+					name,
+					time,
+					link,
+					featured_photo
+				}
+			})
+			.map( (mEvent) => {
```

Comment:
You could consider to make a seperate function for this `map` to make it more readable. But it's not that important.